composer/loggers/in_memory_logger.py

```diff
@@ -74,3 +75,50 @@ def log_metric(self, timestamp: Timestamp, log_level: LogLevel, data: TLogData):
             self.data[k].append((timestamp, log_level, v))
         self.most_recent_values.update(data)
         self.most_recent_timestamps.update({k: timestamp for k in data})
+
+    def get_timeseries(self, metric: str, time_unit: Union[str, None] = None):
+        """Returns logged data as dict containing values of a desired metric over time.
+
+        Args:
+            metric (str): Metric of interest. Must be present in self.data.keys().
+            time_unit(str or None, optional) Which unit of time to return. Options =
+                {"batch", "epoch", "sample", "token", "duration", None}. If ``None``,
+                will return all available units of time. Default=``None``.
+
+        Returns:
+            timeseries (dict): Dictionary in which one key is ``metric``, and the
+                associated value is a list of values of that metric. The remaining key(s)
+                are each a unit of time, and the associated values are each a list of
+                values of that time unit for the corresponding index of the metric. For
+                example, plotting timeseries['accuracy/val'] vs. timeseries['epoch'] would
+                plot validation accuracy as a function of epoch. And if
+                timeseries['accuracy/val'][5] == 41.2, and timeseries['batch'][5] == 680,
+                that means that 'accuracy/val' was equal to 41.2 on the 680th batch.
+        """
+
+        # Check time_unit string
+        valid_time_strings = {"batch", "epoch", "sample", "token", "duration", None}
+        if time_unit not in valid_time_strings:
```

Comment:
you can try casting via `time_unit = TimeUnit(time_unit)` -- if it's an illegal unit, then a ValueError will be raised. So no need to validate manually,

Reply:
`TimeUnit` allows for `duration`, but `TimeStamp` does not have the `duration` field.

Reply:
If it's just one value that's not supported, can we manually check for that and raise an error? Not a fan of manually repeating this here.